src/esp_littlefs.c

```diff
@@ -943,7 +1101,11 @@ static int vfs_littlefs_open(void* ctx, const char * path, int flags, int mode)
     ESP_LOGV(ESP_LITTLEFS_TAG, "Opening %s", path);
 
     /* Convert flags to lfs flags */
-    lfs_flags = esp_littlefs_flags_conv(flags);
+    if(efs->read_only) {
+        lfs_flags = LFS_O_RDONLY;
+    } else {
+        lfs_flags = esp_littlefs_flags_conv(flags);
+    }
```

Comment:
I think this should be:

```
    if(efs->read_only) {
        assert(lfs_flags == LFS_O_RDONLY);
    }
    lfs_flags = esp_littlefs_flags_conv(flags);
```

That way it forces the caller to pass in the correct flag.

Reply:
Hmm, that will cause the assert to happen on lfs_flags before it is initialized. Don't think that will work. I can move the check below the `lfs_flags = esp_littlefs_flags_conv(flags);` line, but that will make some of the tests i have fail as they actually open files in write mode and tests that they can't be written to. I can remove those tests, but i personally think the better approach is to force the mode to read-only when the filesystem has been mounted as read-only

Reply:
ok, you're right, it should be more like:

```
    lfs_flags = esp_littlefs_flags_conv(flags);
    if(efs->read_only) {
        assert(lfs_flags == LFS_O_RDONLY);
    }
```

And then maybe, instead of the `assert`, we could do the following:
```
    if(efs->read_only && lfs_flags != LFS_O_RDONLY) {
        return LFS_ERR_INVAL;
    }
```

Basically, I don't think someone should be able to call `open("blah", "w");` whenever the filesystem is readonly (instead of us silently changing the mode to `"r"`).

Reply:
I like the last suggestion, then fopen will just return NULL if trying to open a file in write mode. Updated the branch.